app/apollo/test/channel.fga.spec.js

```diff
@@ -0,0 +1,666 @@
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user01.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser01Data);
+  fgaUser02Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user02.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser02Data);
+  return {};
+};
+
+const createChannels = async () => {
+  await models.Channel.create({
+    _id: 'fake_ch_id_1',
+    org_id: org01._id,
+    uuid: 'testConfiguration1',
+    name: 'test-configuration1',
+    versions: [],  /* channel versions is deprecated and no longer used */
+    tags: ['tag_01'],
+    contentType: 'local'
+  });
+  await models.Channel.create({
+    _id: 'fake_ch_id_2',
+    org_id: org01._id,
+    uuid: 'testConfiguration2',
+    name: 'test-configuration2',
+    versions: [],  /* channel versions is deprecated and no longer used */
+    tags: ['tag_02'],
+    contentType: 'local'
+  });
+};
+
+const createVersions = async () => {
+  await models.DeployableVersion.create({
+    _id: 'fake_ver_id_1',
+    org_id: org01._id,
+    uuid: 'testVersion1',
+    name: 'test-version1',
+    channel_id: 'testConfiguration1',
+    channel_name: 'test-configuration1',
+  });
+  await models.DeployableVersion.create({
+    _id: 'fake_ver_id_2',
+    org_id: org01._id,
+    uuid: 'testVersion2',
+    name: 'test-version2',
+    channel_id: 'testConfiguration2',
+    channel_name: 'test-configuration2',
+  });
+};
+
+const createGroups = async () => {
+  await models.Group.create({
+    _id: UUID(),
+    org_id: org01._id,
+    uuid: 'testGroup1',
+    name: 'testGroup1',
+    owner: 'undefined'
+  });
+  await models.Group.create({
+    _id: UUID(),
+    org_id: org01._id,
+    uuid: 'testGroup2',
+    name: 'testGroup2',
+    owner: 'undefined'
+  });
+};
+
+describe('channel graphql test suite', () => {
+  before(async () => {
+    process.env.NODE_ENV = 'test';
+    mongoServer = new MongoMemoryServer( { binary: { version: '4.2.17' } } );
+    await mongoServer.start();
+    const mongoUrl = mongoServer.getUri();
+    console.log(`    cluster.js in memory test mongodb url is ${mongoUrl}`);
+
+    myApollo = await apollo({
+      mongo_url: mongoUrl,
+      graphql_port: graphqlPort,
+    });
+
+    await createOrganizations();
+    await createUsers();
+    await createChannels();
+    await createVersions();
+    await createGroups();
+
+    fgaToken01 = await signInUser(models, resourceApi, fgaUser01Data);
+    fgaToken02 = await signInUser(models, resourceApi, fgaUser02Data);
+  }); // before
+
+  after(async () => {
+    await myApollo.stop(myApollo);
+    await mongoServer.stop();
+  }); // after
+
+  // channels fgaUser01
+  it('fgaUser01 has authentication to get ALLOWED channels', async () => {
+    try {
+      const {
+        data: {
+          data: { channels },
+        },
+      } = await channelApi.channels(fgaToken01, {
+        orgId: org01._id,
+      });
+
+      expect(channels).to.have.length(1);
```

Comment:
This code pattern will make debugging test failures very difficult, e.g. would typically require recreating the failure locally and then modifying the test code to print out more debug info.  This is because an error will be thrown when trying to set the `channels` variable if the response does not include `data.data.channels` -- the error handling will catch the thrown error and log something like `error encountered:   unable to find attribute of undefined`, but it won't explain what the response actually _**was**_.  All you'd know is that it apparently didn't have some part of `data.data.channels`.

Instead the pattern needs to be more like (but not necessarily identical to) https://github.com/razee-io/Razeedash-api/blob/master/app/apollo/test/cluster.spec.js#L923-L929, where it sets a `response` variable, _logs it_, and then checks the response with `expect` statements.  Possible improvements on top of even that example would be to only log the response details if there's an error, e.g. something like:
```
    let response;
    try {
      response = await exampleApi.exampleFunc(token, {
        exampleArg: 'exampleVal',
      });
      expect(response.data.data.exampleFunc.exampleAttr).to.be.an('string');
    } catch (error) {
      console.error('API response: ', response);
      console.error('Test failure, error: ', error);
      throw error;
    }
```

The code pattern chosen should also be used consistently for all tests -- the `// channels fgaUser02` test immediately below this one should be almost identical (it's the exact same test as this one, just for `fgaUser02` instead of `fgaUser01`), but it's using a different code pattern (which also has the same problems of debug-ability):
```
      const result = await channelApi.channels(fgaToken02, {
        orgId: org01._id,
      });

      expect(result.data.data.channels).to.have.length(1);
```

Reply:
Since this test is verifying listing channels, it should verify the returned channel is the _correct_ one, not just that it only returned one.  E.g. something like
```
expect(result.data.data.channels).to.have.length(1);
expect(result.data.data.channels[0].name).to.equal('test-configuration1');
```
Same for the `fgaUser02` test following (with `2` name natch)

Reply:
Completed for channel.fga.spec.js

Reply:
Resolved (in channel.fga.spec.js at least, others pending)

Reply:
Completed for all fga.spec.js files

Reply:
resolved